com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainClassHandler.java

```diff
@@ -185,23 +189,41 @@ private ValidationResponse validateLaunchConfigCore(List<Object> arguments) thro
     private ValidationResult validateMainClass(final String mainClass, final String projectName, boolean containsExternalClasspaths) throws CoreException {
         if (StringUtils.isEmpty(mainClass)) {
             return new ValidationResult(true);
-        } else if (!mainClass.matches(CLASSNAME_REGX)) {
-            return new ValidationResult(false, String.format("ConfigError: '%s' is not a valid class name.", mainClass));
+        } else if (!isValidMainClassName(mainClass)) {
+            return new ValidationResult(false, String.format("ConfigError: '%s' is not a valid class name.", mainClass),
+                CONFIGERROR_INVALID_CLASS_NAME);
         }
 
         if (!containsExternalClasspaths && StringUtils.isEmpty(projectName)) {
             List<IJavaProject> javaProjects = searchClassInProjectClasspaths(mainClass);
             if (javaProjects.size() == 0) {
-                return new ValidationResult(false, String.format("ConfigError: Main class '%s' doesn't exist in the workspace.", mainClass));
+                return new ValidationResult(false, String.format("ConfigError: Main class '%s' doesn't exist in the workspace.", mainClass),
+                    CONFIGERROR_MAIN_CLASS_NOT_EXIST);
             }
             if (javaProjects.size() > 1) {
-                return new ValidationResult(false, String.format("ConfigError: Main class '%s' isn't unique in the workspace.", mainClass));
+                return new ValidationResult(false, String.format("ConfigError: Main class '%s' isn't unique in the workspace.", mainClass),
+                    CONFIGERROR_MAIN_CLASS_NOT_UNIQUE);
             }
         }
 
         return new ValidationResult(true);
     }
 
+    // Java command line supports two kinds of main class format: <mainclass> and <module>[/<mainclass>]
+    private boolean isValidMainClassName(String mainClass) {
+        if (StringUtils.isEmpty(mainClass)) {
```

Comment:
Curious what will happen if main class name is empty?

Reply:
when main class is not specified (e.g. empty), the validator will pass. because the debugger will auto resolve them from search engine once it's not configured by user.

Reply:
I see..